x-pack/plugins/security_solution/common/api/detection_engine/prebuilt_rules/perform_rule_upgrade/perform_rule_upgrade_route.schema.yaml

```diff
@@ -0,0 +1,507 @@
+                  $ref: '../../model/rule_schema/specific_attributes/threat_match_attributes.schema.yaml#/components/schemas/ThreatMapping'
+            threat_index:
+              type: object
+              required:
+                - pick_version
+              properties:
+                pick_version:
+                  $ref: '#/components/schemas/FieldPickVersionValues'
+                resolved_value:
+                  $ref: '../../model/rule_schema/specific_attributes/threat_match_attributes.schema.yaml#/components/schemas/ThreatIndex'
+            threat_filters:
+              type: object
+              required:
+                - pick_version
+              properties:
+                pick_version:
+                  $ref: '#/components/schemas/FieldPickVersionValues'
+                resolved_value:
+                  $ref: '../../model/rule_schema/specific_attributes/threat_match_attributes.schema.yaml#/components/schemas/ThreatFilters'
+            threat_indicator_path:
+              type: object
+              required:
+                - pick_version
+              properties:
+                pick_version:
+                  $ref: '#/components/schemas/FieldPickVersionValues'
+                resolved_value:
+                  $ref: '../../model/rule_schema/specific_attributes/threat_match_attributes.schema.yaml#/components/schemas/ThreatIndicatorPath'
+            threat_language:
+              type: object
+              required:
+                - pick_version
+              properties:
+                pick_version:
+                  $ref: '#/components/schemas/FieldPickVersionValues'
+                resolved_value:
+                  $ref: '../../model/rule_schema/common_attributes.schema.yaml#/components/schemas/KqlQueryLanguage'
+            new_terms_fields:
+              type: object
+              required:
+                - pick_version
+              properties:
+                pick_version:
+                  $ref: '#/components/schemas/FieldPickVersionValues'
+                resolved_value:
+                  $ref: '../../model/rule_schema/specific_attributes/new_terms_attributes.schema.yaml#/components/schemas/NewTermsFields'
+            history_window_start:
+              type: object
+              required:
+                - pick_version
+              properties:
+                pick_version:
+                  $ref: '#/components/schemas/FieldPickVersionValues'
+                resolved_value:
+                  $ref: '../../model/rule_schema/specific_attributes/new_terms_attributes.schema.yaml#/components/schemas/HistoryWindowStart'
+
+    UpgradeSpecificRulesRequest:
+      type: object
+      required:
+        - mode
+        - rules
+      properties:
+        mode:
+          type: string
+          enum: [SPECIFIC_RULES]
+        rules:
+          type: array
+          items:
+            $ref: '#/components/schemas/RuleUpgradeSpecifier'
+        pick_version:
```

Comment:
How would `pick_version` specified at different levels work? 
Will rule-level `pick_version` override root-level `pick-version`? 
Will field-level `pick_version` override both rule-level and root-level `pick_version`?

Reply:
Yes, more specific levels will override higher level ones, when defined.